src/configure/configure.ts

```diff
@@ -2,41 +2,58 @@ const uuid = require('uuid/v4');
 import { AppServiceClient } from './clients/azure/appServiceClient';
 import { AzureDevOpsClient } from './clients/devOps/azureDevOpsClient';
 import { AzureDevOpsHelper } from './helper/devOps/azureDevOpsHelper';
-import { AzureTreeItem } from 'vscode-azureextensionui';
+import { AzureTreeItem, UserCancelledError } from 'vscode-azureextensionui';
 import { exit } from 'process';
 import { generateDevOpsProjectName, generateDevOpsOrganizationName } from './helper/commonHelper';
 import { GenericResource } from 'azure-arm-resource/lib/resource/models';
 import { GraphHelper } from './helper/graphHelper';
 import { LocalGitRepoHelper } from './helper/LocalGitRepoHelper';
-import { Messages } from './messages';
+import { Messages } from './resources/messages';
 import { QuickPickItem } from 'vscode';
 import { ServiceConnectionHelper } from './helper/devOps/serviceConnectionHelper';
 import { SourceOptions, RepositoryProvider, extensionVariables, WizardInputs, WebAppKind, PipelineTemplate, QuickPickItemWithData } from './model/models';
+import { TracePoints } from './resources/tracePoints';
+import { TelemetryKeys } from './resources/telemetryKeys';
 import * as path from 'path';
 import * as templateHelper from './helper/templateHelper';
 import * as utils from 'util';
 import * as vscode from 'vscode';
+import { TelemetryHelper } from './helper/telemetryHelper';
+import { ControlProvider } from './helper/controlProvider';
 
-export async function configurePipeline(node: any) {
-    try {
-        if (!(await extensionVariables.azureAccountExtensionApi.waitForLogin())) {
-            let signIn = await vscode.window.showInformationMessage(Messages.azureLoginRequired, Messages.signInLabel);
-            if(signIn.toLowerCase() === Messages.signInLabel.toLowerCase()) {
-                await vscode.commands.executeCommand("azure-account.login");
+const Layer: string = 'configure';
+
+export async function configurePipeline(telemetryHelper: TelemetryHelper, node: AzureTreeItem) {
+    await telemetryHelper.execteFunctionWithTimeTelemetry(async () => {
+        try {
+            if (!(await extensionVariables.azureAccountExtensionApi.waitForLogin())) {
+                // set telemetry
+                telemetryHelper.setTelemetry(TelemetryKeys.AzureLoginRequired, 'true');
+
+                let signIn = await vscode.window.showInformationMessage(Messages.azureLoginRequired, Messages.signInLabel);
+                if (signIn.toLowerCase() === Messages.signInLabel.toLowerCase()) {
+                    await vscode.commands.executeCommand("azure-account.login");
+                }
+                else {
+                    let error = new Error(Messages.azureLoginRequired);
+                    telemetryHelper.setResult('Failed', error);
```

Comment:
Do you want to use Enum for result as you are calling setResult multiple times with result as string ?

Reply:
The parameter accepts only 3 strings hence it gives a compilation error if you pass any string.
Also, if you can help me in converting enum to string, such that compilation error doesn't come, that will be great

Reply:
I mean setResult can take an enum as argument, something like Result = { Failed="Failed", Success = "Success" }